### PR TITLE
Make the rpc package work on Dart 2

### LINKED ISSUE #123: Issues compiling with Dev Sdk
Currently on `2.0.0-dev.67.0` of the sdk. 

A few type errors occur when trying to compile an RPC project. 

Here are some of the common errors that seem to occur.  
```
Error: A value of type '(dart.core::String, dynamic) → dart.core::Null' can't be assigned to a variable of type '(dynamic, dynamic) → void'.
Try changing the type of the left hand side, or casting the right hand side to '(dynamic, dynamic) → void'.
    (mapObject as Map).forEach((String key, object) {

Error: A value of type '(dart.core::String, dynamic) → dart.core::Null' can't be assigned to a variable of type '(dynamic, dynamic) → void'.
Try changing the type of the left hand side, or casting the right hand side to '(dynamic, dynamic) → void'.
    (encodedMap as Map).forEach((String key, encodedObject) {

Error: A value of type '(dart.core::String, dynamic) → dart.core::Null' can't be assigned to a variable of type '(dynamic, dynamic) → void'.
Try changing the type of the left hand side, or casting the right hand side to '(dynamic, dynamic) → void'.
    (result as Map).forEach((String key, value) {

The getter 'isRegularMethod' isn't defined for the class 'dart.mirrors::DeclarationMirror'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'isRegularMethod'.
      if (dm is! MethodMirror || !dm.isRegularMethod) {
```

Seems like most of these errors are a result of the dart strong mode changes for 2.0.

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #123 and #121.

Most of the changes are the usual type inference adjustments/overrides and constant renaming.  However, here are a few highlights of package-specific changes made here:

- pubspec.yaml:  Changes to external package versions to correct problems with our dependencies include convert, discoveryapis_generator, uri, and test
- README.md, lib/src/parser.dart, lib/src/config/property.dart, api_property_int_test.dart:  These files contain the bulk of the non-backwards compatible change to 64 bit integer handling and its documentation.
- bin/generator.dart:  Stopped hiding exception backtraces here because that makes debugging the generator from user reports harder.
- .travis.yml, tool/travis.sh:  New travis configuration for rpc so we'll have continuous integration.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: dart
+sudo: false
```
Comment (kevmoo) [inline .travis.yml]: not needed - sudo is false by default

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+dart:
+  - "dev/raw/latest"
```
Comment (kevmoo) [inline .travis.yml]: Uh...t'would be better if we were consistent here and just stuck w/ `-dev` I think

File: pubspec.yaml [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
 author: Dart Team <misc@dartlang.org>
 description: Library for exposing Dart server-side APIs.
 homepage: https://github.com/dart-lang/rpc
 environment:
-  sdk: '>=1.9.0 <2.0.0'
+  sdk: '>=2.0.0-dev.69.0 <3.0.0'
 dependencies:
   collection: '>=1.1.1 <2.0.0'
-  convert: '>=1.0.0 <3.0.0'
+  convert: '>=2.0.1 <3.0.0'
```
Comment (kevmoo) [inline pubspec.yaml]: Migrate to `^` syntax here

File: pubspec.yaml [hunk 4]
```diff
@@ -1,20 +1,20 @@
...
   crypto: '>=1.0.0 <3.0.0'
-  discoveryapis_generator: '>=0.7.1+1 <0.9.8'
+  discoveryapis_generator: '>=0.9.8 <0.10.0'
```
Comment (kevmoo) [inline pubspec.yaml]: Migrate to `^` syntax here

File: tool/travis.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+# BSD-style license that can be found in the LICENSE file.
+
+# Fast fail the script on failures, and echo commands as they execute.
+set -ex
+
+# add globally activated packages to the path
+export PATH="$PATH":"~/.pub-cache/bin"
+
+pub get
+pub run test:test
```
Comment (kevmoo) [inline tool/travis.sh]: Why not use `darttask:`? https://docs.travis-ci.com/user/languages/dart#Running-Tests

File: lib/src/config/property.dart [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
 part of rpc.config;
 
-class ApiConfigSchemaProperty {
+class ApiConfigSchemaProperty<T> {
```
Comment (mkustermann) [inline lib/src/config/property.dart]: You could consider making 2 type parameters: `D` for the type used in Dart code and `J` for the type used in the json messsages.

Comment (jcollins-g) [reply lib/src/config/property.dart]: Good idea, done.  Also renamed the methods to something that makes a bit more sense to me given their usage (there's not actually a multiple to single conversion going on anywhere I can see, it seems instead that these are just wrapped in a null detector).

File: lib/src/config/property.dart [hunk 7]
```diff
@@ -80,12 +80,16 @@ class IntegerProperty extends ApiConfigSchemaProperty {
             apiType,
             apiFormat);
 
-  _singleResponseValue(value) {
+  _singleResponseValue(dynamic value) {
     assert(value != null);
-    if (value is! int) {
+    if (_apiFormat.endsWith('64') && value is! String && value is! BigInt) {
```
[the remark this answers is not in the document]
Comment (jcollins-g) [reply lib/src/config/property.dart]: Right, this isn't needed, I think it was leftover while I was hacking on things.  Removed.

File: pubspec.yaml (hunk 3 above)
Comment (jcollins-g) [reply pubspec.yaml]: done

File: pubspec.yaml (hunk 4 above)
Comment (jcollins-g) [reply pubspec.yaml]: done

File: tool/travis.sh (hunk 5 above)
Comment (jcollins-g) [reply tool/travis.sh]: no particular reason.  changed.

**Pull request merged**